src/namespace.js

```diff
@@ -0,0 +1,81 @@
+
+export const namespaceSeparator = '/'
+
+export function validateNamespace (namespace) {
+  if (typeof namespace === 'string') {
+    namespace = namespace.split(namespaceSeparator)
+  }
+
+  if (!Array.isArray(namespace)) {
+    // TODO: warn?
+    return []
+  }
+
+  return namespace.filter(Boolean)
+}
+
+export function getPropertyByNamespace (object, namespace) {
```

Comment:
The crux of the methods in this module is to traverse an object by the individual paths of the namespace.